### PR TITLE
Add support for HACS repository redirect

### PR DESCRIPTION
**Pull request by ludeeus**
https://user-images.githubusercontent.com/15093472/177651152-a27a24f4-43ca-47c3-9535-dc5f7b56a68f.mp4

### CODE REVIEW

File: redirect.json [hunk 1]
```diff
@@ -173,6 +173,24 @@
     "description": "show your general Home Assistant settings",
     "introduced": "2021.3"
   },
+  {
+    "redirect": "hacs_repository",
+    "component": "hacs",
+    "name": "HACS: Repository",
+    "badge": "HACS: Repository",
+    "description": "Show the repository inside HACS",
```
Comment (ludeeus) [inline redirect.json:181]:
```suggestion
    "description": "show the repository inside HACS",
```

File: src/entrypoints/my-redirect.ts [hunk 2]
```diff
@@ -65,6 +65,8 @@ const render = (showTroubleshooting: boolean) => {
   const redirectUrl =
     window.redirect.redirect === "oauth"
       ? `${instanceUrl}/auth/external/callback${params}`
+      : window.redirect.custom
+      ? `${instanceUrl}/${window.redirect.component}/_my_redirect/${window.redirect.redirect}${params}`
```
Comment (bramkragten) [inline src/entrypoints/my-redirect.ts:69]: Shouldn't we link to the my integration so we can also catch none installed components? And have more control?

Comment (ludeeus) [reply src/entrypoints/my-redirect.ts:69]: Actually, no I'm not fine with that.
a) We would not get more control (we would still require redirects to register in this repo).
b) The codebase of custom integrations does not have to follow the pase of core and vise-versa, which will make rolling out new paths much slower, and would require a lot of effort on all parts to make it compatible with as wide of a version span as possible.
c) Sure we get a component not loaded check, but nothing else, and with a and b, I don't see that single item way up against it.

File: src-11ty/redirect-page.html [hunk 3]
```diff
@@ -50,6 +50,11 @@
   .highlight {
     display: none;
   }
+  .warning {
```
Comment (balloob) [inline src-11ty/redirect-page.html:53]: Let's copy the warning UI from HA frontend.

File: src-11ty/redirect-page.html [hunk 4]
```diff
@@ -63,6 +68,11 @@ <h1 class="card-header">Link account to Home Assistant?</h1>
   <h1 class="card-header">Open page in your Home Assistant?</h1>
   <div class="card-content">
     You've been linked to the page that will {{ redirect.description }}.
+    {% if redirect.custom %}
+    <div class="warning">
+      Warning!: This link redirects to a custom integration.
```
Comment (balloob) [inline src-11ty/redirect-page.html:73]: `!:` is weird.

File: src/entrypoints/my-redirect.ts (hunk 2 above)
Comment (bramkragten) [reply src/entrypoints/my-redirect.ts:69]: Extra control in that we could add an extra warning, check if the link works, check for versions of integrations, etc.

That rolling out new paths will be slower is not that big of an issue is it? Do we plan to add a lot of these?

File: redirect.json [hunk 5]
```diff
@@ -173,6 +173,24 @@
     "description": "show your general Home Assistant settings",
     "introduced": "2021.3"
   },
+  {
+    "redirect": "hacs_repository",
+    "component": "hacs",
+    "name": "HACS: Repository",
+    "badge": "HACS: Repository",
+    "description": "show the repository inside HACS",
```
Comment (balloob) [inline redirect.json:181]:
```suggestion
    "description": "open a repository inside the Home Assistant Community Store",
```

Comment (balloob) [reply redirect.json:181]: btw `repository` is a very technical term.

File: src/entrypoints/my-redirect.ts (hunk 2 above)
Comment (balloob) [reply src/entrypoints/my-redirect.ts:69]: People are going to end up clicking on a badge, seeing official My Home Assistant page and end up in their instance with a 404 error page without any UI:

image

The My panel can handle this:

image

Comment (ludeeus) [reply src/entrypoints/my-redirect.ts]: https://github.com/home-assistant/frontend/pull/13153

File: src-11ty/redirect-page.html [hunk 6]
```diff
@@ -63,6 +89,14 @@ <h1 class="card-header">Link account to Home Assistant?</h1>
   <h1 class="card-header">Open page in your Home Assistant?</h1>
   <div class="card-content">
     You've been linked to the page that will {{ redirect.description }}.
+    {% if redirect.custom %}
```
Comment (balloob) [inline src-11ty/redirect-page.html:92]: Don't put it inside the card-content. I think above the header is better? (but kill the white space between blue top border and the warning)

https://deploy-preview-256--my-home-assistant.netlify.app/redirect/hacs_repository/?owner=asd&repository=asd

### Current

image

### Moved higher

image


Also looks like there needs to be more space between icon and text

**Pull request merged**